parcels/grid.py

```diff
@@ -0,0 +1,142 @@
+from parcels.loggers import logger
+import numpy as np
+from ctypes import Structure, c_int, c_float, c_double, POINTER, cast, c_char_p, c_void_p, pointer
+from enum import IntEnum
+
+__all__ = ['GridCode', 'StructuredGrid', 'GridIndex', 'CGrid']
+
+
+class GridCode(IntEnum):
+    StructuredGrid = 0
+    SemiStructuredGrid = 1
+
+
+class CGrid(Structure):
+    _fields_ = [('name', c_char_p),
+                ('gtype', c_int),
+                ('grid', c_void_p)]
+
+
+class Grid(object):
+    """Grid class that defines a (spatial and temporal) grid on which Fields are defined
+
+    """
+
+    @property
+    def ctypes_struct(self):
+        self.cgrid = cast(pointer(self.child_ctypes_struct), c_void_p)
+        cstruct = CGrid(self.name, self.gtype, self.cgrid.value)
+        return cstruct
+
+
+class StructuredGrid(Grid):
+    """Structured Grid
+
+    :param name: Name of the field
+    :param lon: Longitude coordinates of the field
+    :param lat: Latitude coordinates of the field
+    :param depth: Depth coordinates of the field
+    :param time: Time coordinates of the field
```

Comment:
Make explicit here in the docstring that `lon`, `lat` and `depth` can be 1D, 2D or 3D arrays, but that time has to be 1D?